ctru-rs/src/services/apt.rs

```diff
@@ -90,3 +132,48 @@ impl Drop for Apt {
         unsafe { ctru_sys::aptExit() };
     }
 }
+
+/// Can launch other applications when the current one exits.
+pub struct Chainloader<'a> {
```

Comment:
`Chainloader` should clear on `Drop`, so that leftovers from previous usages won't influence in un-mutable environments.

Reply:
i did think about this, but then forgot to propose it.

the problem is that the destructor will run when you exit the app, as the chainloader goes out of scope.

Reply:
Ah right, didn’t think about it.

Wait, does this mean the setup only works because there is a reference to `Apt` that doesn’t get uninitialised? (The one made in libctru, i mean). If this were to not work without that (because the apt service were to be unused) then we *must* find a way around this.

Reply:
at a first look it definitely looks that way.